posthog/hogql_queries/experiment_funnel_query_runner.py

```diff
@@ -17,16 +21,51 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.experiment = Experiment.objects.get(id=self.query.experiment_id)
         self.feature_flag = self.experiment.feature_flag
-
+        self.prepared_funnel_query = self._prepare_funnel_query()
         self.query_runner = FunnelsQueryRunner(
-            query=self.query.source, team=self.team, timings=self.timings, limit_context=self.limit_context
+            query=self.prepared_funnel_query, team=self.team, timings=self.timings, limit_context=self.limit_context
         )
 
     def calculate(self) -> ExperimentFunnelQueryResponse:
         response = self.query_runner.calculate()
         results = self._process_results(response.results)
         return ExperimentFunnelQueryResponse(insight="FUNNELS", results=results)
 
+    def _prepare_funnel_query(self) -> FunnelsQuery:
+        """
+        This method takes the raw funnel query and adapts it
+        for the needs of experiment analysis:
+
+        1. Set the date range to match the experiment's duration, using the project's timezone.
+        2. Configure the breakdown to use the feature flag key, which allows us
+           to separate results for different experiment variants.
+        """
+        # Clone the source query
+        prepared_funnel_query = FunnelsQuery(**self.query.source.model_dump())
+
+        # Set the date range to match the experiment's duration, using the project's timezone
+        if self.team.timezone:
+            tz = ZoneInfo(self.team.timezone)
+            start_date = self.experiment.start_date.astimezone(tz) if self.experiment.start_date else None
+            end_date = self.experiment.end_date.astimezone(tz) if self.experiment.end_date else None
```

Comment:
is `None` an typically expected outcome, or do most queries tend to have time ranges on them (and this else is just a default case that isn't common)?